lib/Dialect/TritonGPU/Transforms/Coalesce.cpp

```diff
@@ -95,6 +95,12 @@ struct CoalescePass : public impl::TritonGPUCoalesceBase<CoalescePass> {
       perThread = std::min<int>(
           perThread, getNumElementsPerThread(op, order, axisInfoAnalysis));
     }
+
+    if (perThread == 1) {
+      op->emitRemark()
```

Comment:
would be good to have something more actionable. I don't think performance warnings are super useful if it doesn't at least point to the root line of code where things went wrong

Reply:
My current rough idea is to report some information about the shapes or axis info, which causes the size per thread becomes 1. Users can then change the shape of tensors.